src/macaron/repo_finder/provenance_finder.py

```diff
@@ -0,0 +1,232 @@
+            return None
+
+        # The size of the asset (in bytes) is added to match the AssetLocator
+        # protocol and is not used because npm API registry does not provide it, so it is set to zero.
+        npm_provenance_asset = NPMAttestationAsset(
+            namespace=namespace,
+            artifact_id=artifact_id,
+            version=version,
+            npm_registry=npm_registry,
+            size_in_bytes=0,
+        )
+        try:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                download_path = os.path.join(temp_dir, f"{artifact_id}.intoto.jsonl")
+                if not npm_provenance_asset.download(download_path):
+                    logger.debug("Unable to find an npm provenance for %s@%s", artifact_id, version)
+                    return None
+
+                try:
+                    # Load the provenance file.
+                    provenance_payload = load_provenance_payload(download_path)
+                except LoadIntotoAttestationError as loadintotoerror:
+                    logger.error("Error while loading provenance %s", loadintotoerror)
+                    return None
+
+                return provenance_payload
+        except OSError as error:
+            logger.error("Error while storing provenance in the temporary directory: %s", error)
+            return None
+
+    @staticmethod
+    def find_gav_provenance(purl: PackageURL) -> InTotoPayload | None:
+        """Find and download the GAV based provenance for the passed PURL.
+
+        Parameters
+        ----------
+        purl: PackageURL
+            The PURL of the analysis target.
+
+        Returns
+        -------
+        InTotoPayload | None
+            The provenance payload if found, or None.
+
+        """
+        jfrog_section = "package_registry.jfrog.maven"
+        if not defaults.has_section(jfrog_section):
+            logger.debug("No JFrog section found in config.")
+            return None
+
+        try:
+            request_timeout = defaults.getint(jfrog_section, "request_timeout")
+            download_timeout = defaults.getint(jfrog_section, "download_timeout")
+        except ValueError as error:
+            logger.debug("Failed to parse default value as int: %s", error)
+            return None
+
+        jfrog_registry = JFrogMavenRegistry(
+            defaults.get(jfrog_section, "hostname"),
+            defaults.get(jfrog_section, "repo"),
+            request_timeout,
+            download_timeout,
+        )
+
+        provenance_extensions = defaults.get_list(
+            "slsa.verifier",
+            "provenance_extensions",
+            fallback=["intoto.jsonl"],
+        )
+
+        provenance_assets = jfrog_registry.fetch_assets(
+            group_id=purl.namespace if purl.namespace else "",
+            artifact_id=purl.name,
+            version=purl.version if purl.version else "",
+            extensions=set(provenance_extensions),
+        )
```

Comment:
I don't think the `.fetch_assets` method would accept empty strings for `group_id` and `version`. It is probably better to check if they are empty and stop proceeding before calling the method.

I designed the API of the `JFrogMavenRegistry` class to be: if an argument is required, it is assumed to be a valid non-empty string. Otherwise, the argument should be optional, and takes `str | None`.